crates/sui-core/src/authority/consensus_tx_status_cache.rs

```diff
@@ -0,0 +1,316 @@
+
+        // Set initial status
+        cache.set_transaction_status(tx_pos, ConsensusTxStatus::FastpathCertified);
+
+        // Read status immediately
+        let result = cache.notify_read_transaction_status(tx_pos, None).await;
+        assert!(matches!(
+            result,
+            NotifyReadConsensusTxStatusResult::Status(ConsensusTxStatus::FastpathCertified)
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_status_notification() {
+        let cache = Arc::new(ConsensusTxStatusCache::new(GC_DEPTH));
+        let tx_pos = create_test_tx_position(1, 0);
+
+        // Spawn a task that waits for status update
+        let cache_clone = cache.clone();
+        let handle = tokio::spawn(async move {
+            cache_clone
+                .notify_read_transaction_status(tx_pos, None)
+                .await
+        });
+
+        // Small delay to ensure the task is waiting
+        tokio::time::sleep(Duration::from_millis(10)).await;
+
+        // Set the status
+        cache.set_transaction_status(tx_pos, ConsensusTxStatus::Finalized);
+
+        // Verify the notification was received
+        let result = handle.await.unwrap();
+        assert!(matches!(
+            result,
+            NotifyReadConsensusTxStatusResult::Status(ConsensusTxStatus::Finalized)
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_round_expiration() {
+        let cache = ConsensusTxStatusCache::new(GC_DEPTH);
+        let tx_pos = create_test_tx_position(1, 0);
+
+        // Set initial status
+        cache.set_transaction_status(tx_pos, ConsensusTxStatus::FastpathCertified);
+
+        // Update last committed round to trigger expiration
+        cache.update_last_committed_round(GC_DEPTH + 2).await;
+
+        // Try to read status - should be expired
+        let result = cache.notify_read_transaction_status(tx_pos, None).await;
+        assert!(matches!(
+            result,
+            NotifyReadConsensusTxStatusResult::Expired(_)
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_multiple_status_updates() {
+        let cache = ConsensusTxStatusCache::new(GC_DEPTH);
+        let tx_pos = create_test_tx_position(1, 0);
+
+        // Set initial status
+        cache.set_transaction_status(tx_pos, ConsensusTxStatus::FastpathCertified);
+
+        // Update status
+        cache.set_transaction_status(tx_pos, ConsensusTxStatus::Finalized);
+
+        // Read with old status
+        let result = cache
+            .notify_read_transaction_status(tx_pos, Some(ConsensusTxStatus::FastpathCertified))
+            .await;
+        assert!(matches!(
+            result,
+            NotifyReadConsensusTxStatusResult::Status(ConsensusTxStatus::Finalized)
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_cleanup_expired_rounds() {
+        let cache = ConsensusTxStatusCache::new(GC_DEPTH);
+
+        // Add transactions for multiple rounds
+        for round in 1..=5 {
```

Comment:
Maybe you could add some transactions for a few more rounds which should not be GC'ed . Then you assert that only the desired rounds have been cleaned up - rest should not.

Reply:
That is exactly what this test did though.

Reply:
oh god. Apologies, I didn't notice the `!` on the last two checks. Ignore.